### PR TITLE
Add mmg

### PR DESCRIPTION
**Pull request by updega2**
Branch to merge in new tools from mmg. 

Builds on mac and linux with downloadable mmg.

Off by default.

### CODE REVIEW

File: Code/SuperBuild/MMG.cmake [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   PREFIX ${${proj}_OUTPUT_DIR}-prefix
+   SOURCE_DIR ${${proj}_OUTPUT_DIR}
+   BINARY_DIR ${${proj}_OUTPUT_BIN_DIR}
+   UPDATE_COMMAND ""
+   CMAKE_CACHE_ARGS
+   -DCMAKE_CXX_COMPILER:STRING=${CMAKE_CXX_COMPILER}
+   -DCMAKE_C_COMPILER:STRING=${CMAKE_C_COMPILER}
+   -DCMAKE_CXX_FLAGS:STRING=${CMAKE_CXX_FLAGS}
+   -DCMAKE_C_FLAGS:STRING=${CMAKE_C_FLAGS}
+   -DCMAKE_THREAD_LIBS:STRING=-lpthread
```
Comment (updega2) [inline Code/SuperBuild/MMG.cmake]: It looks like -lpthread is a flag for VTK and ITK too. Those have been okay for windows so far. I'm going to leave the flag in for the commit. If you want me to remove these for windows for all of them, I will do it in a separate pull request @osmsc

**Comment by osmsc** [comment]
How stable is mmg?  If it's small, does it make sense to put the source into "ThirdParty" or to keep it an external?

**Comment by updega2** [comment]
Its pretty small, you can look at the fork I made for it. I would rather keep it as an external though because it is currently under development, and I would like to be able to update easily with their version releases. If we put it in ThirdParty, that gets more difficult to do.

**Comment by osmsc** [comment]
Ok, we can keep it in external.  We have a few small things in external already (e.g. freetype), so it's not a big deal just wanted to ask.

**Pull request merged**